compiler/src/dotty/tools/backend/jvm/GenBCode.scala

```diff
@@ -66,25 +66,28 @@ class GenBCode extends Phase { self =>
     _postProcessor.nn
   }
 
-  override def run(using ctx: Context): Unit =
-    // CompilationUnit is the only component that will differ between each run invocation
-    // We need to update it to have correct source positions.
-    // FreshContext is always enforced when creating backend interface
-    backendInterface.ctx
-      .asInstanceOf[FreshContext]
-      .setCompilationUnit(ctx.compilationUnit)
-    val generated = codeGen.genUnit(ctx.compilationUnit)
-    // In Scala 2, the backend might use global optimizations which might delay post-processing to build the call graph.
-    // In Scala 3, we don't perform backend optimizations and always perform post-processing immediately.
-    // https://github.com/scala/scala/pull/6057
-    postProcessor.postProcessAndSendToDisk(generated)
+  private var _generatedClassHandler: GeneratedClassHandler | Null = _
+  def generatedClassHandler(using Context): GeneratedClassHandler = {
+    if _generatedClassHandler eq null then
+      _generatedClassHandler = GeneratedClassHandler(postProcessor)
+    _generatedClassHandler.nn
+  }
+
+  override def run(using Context): Unit =
+    frontendAccess.frontendSynch {
+      backendInterface.curCtx = ctx
```

Comment:
This did not change before. Why does it need it to change now? I'm always wary of `using var`s. They are bad smell.

Reply:
Looks like an artifact left after rebase, it's fixed now